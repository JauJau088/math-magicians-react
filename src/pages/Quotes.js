import React from 'react';
import quotes from '../data/quotes';

const Quotes = () => {
  const quote = quotes[Math.floor(Math.random() * quotes.length)];

  return (
    <div>
      <p>{quote.quote}</p>
      <p>
        -&nbsp;
        {quote.name}
        &nbsp;
        (
        {quote.year}
        )
      </p>
      <p>{quote.title}</p>
    </div>
  );
};

export default Quotes;
