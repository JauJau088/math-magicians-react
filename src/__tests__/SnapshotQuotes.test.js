import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from '../pages/Quotes';

describe('Snapshot test for Quotes page', () => {
  it('should render home page identical to the snapshot', () => {
    const quotes = renderer.create(<Quotes />).toJSON();

    // Since the quotes are generated dynamically, we don't need to check the exact match
    // We just need to check if the pattern is the same
    // Uncomment the following to check the why toMatchSnapshot's pattern is built like that
    // console.log('LOG quotes: ', quotes);
    // console.log('LOG CHILDREN: ', quotes.children[0].children[0].children[0]);
    expect(quotes)
      .toMatchSnapshot({ children: [{ children: [{ children: expect.any(Object) }] }] });
  });
});
