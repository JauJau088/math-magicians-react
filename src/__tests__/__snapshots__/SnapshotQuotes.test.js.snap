// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Quotes page should render home page identical to the snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Any<Object>,
          "props": Object {
            "className": "SectionContent",
          },
          "type": "div",
        },
      ],
      "props": Object {
        "className": "ChalkBoard SectionContentContainer",
      },
      "type": "div",
    },
  ],
  "props": Object {
    "className": "SectionContainer",
  },
  "type": "div",
}
`;
